UnityProject/Assets/Scripts/Items/Weapons/Gun.cs

```diff
@@ -366,63 +392,44 @@ public virtual void ServerPerformInteraction(AimApply interaction)
 			gunTrigger.TriggerPull(interaction.Performer, interaction.TargetVector.normalized, UIManager.DamageZone, isSuicide);
 		}
 
-
-
-		public virtual bool Interact(HandActivate interaction)
+		public virtual void ServerPerformInteraction(HandActivate interaction)
 		{
-			//try ejecting the mag if external
 			if (CurrentMagazine != null && allowMagazineRemoval && !MagInternal)
 			{
 				RequestUnload(CurrentMagazine);
-				return true;
 			}
-
-			return false;
 		}
 
 		public void ServerPerformInteraction(InventoryApply interaction)
 		{
-			if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Suppressor) && !isSuppressed && isSuppressible)
-			{
-				isSuppressed = true;
-				// setting the suppressor var to the gameobject and then spawning that would be better,
-				// however the despawning sets the var to null
-				Despawn.ServerSingle(interaction.UsedObject);
-			}
-			else if (isSuppressed && isSuppressible && interaction.UsedObject == null)
-			{
-				isSuppressed = false;
-				var result = Spawn.ServerPrefab(suppressor);
-				if (result.Successful)
-				{
-				var item = result.GameObject;
-				Inventory.ServerAdd(item, interaction.FromSlot);
-				}
-			}
-		}
-
-		public bool WillInteract(InventoryApply interaction, NetworkSide side)
-		{
-			if (!DefaultWillInteract.Default(interaction, side)) return false;
-			if (side == NetworkSide.Server && DefaultWillInteract.Default(interaction, side)) return true;
-			//only reload if the gun is the target and mag/clip is in hand slot
-			if (interaction.TargetObject == gameObject && interaction.IsFromHandSlot && side == NetworkSide.Client)
+			if (interaction.TargetObject == gameObject && interaction.IsFromHandSlot)
 			{
-				if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Suppressor) || interaction.IsAltClick)
-				{
-					return true;
-				}
-				else if (interaction.UsedObject != null)
+				if (interaction.UsedObject != null)
 				{
 					MagazineBehaviour mag = interaction.UsedObject.GetComponent<MagazineBehaviour>();
 					if (mag)
 					{
-						TryReload(mag.gameObject);
-						return true;
+						RequestReload(mag.gameObject);
+					}
+					else if (Validations.HasItemTrait(interaction.UsedObject, CommonTraits.Instance.Suppressor) && !isSuppressed && isSuppressible)
+					{
+						SyncIsSuppressed(isSuppressed, true);
+						// setting the suppressor var to the gameobject and then spawning that would be better,
+						// however the despawning sets the var to null
+						Despawn.ServerSingle(interaction.UsedObject);
+					}
+				}
+				else if (isSuppressed && isSuppressible)
```

Comment:
Not that it will make any difference in performance, but it makes more sense to me to ask if it is suppressible first.